SimMuon/RPCDigitizer/src/RPCSimSetUp.cc

```diff
@@ -166,6 +168,213 @@ void RPCSimSetUp::setRPCSetUp(const std::vector<RPCStripNoises::NoiseItem>& vnoi
+      clsCounter=0;
+    }
+    /*else{
+      LogDebug ("rpssimsetup")<<" --> not saved in map "<<std::endl;
+    }*/
+    ++clsCounter;
+  }
+  // ### loop for Old Format (100 entries)
+  for(itCls = vClusterSize.begin(); itCls != vClusterSize.end(); ++itCls){
+    // LogDebug ("rpssimsetup")<<" Push back clustersize = "<<itCls->clusterSize<<std::endl;
+    clsVect.push_back(((double)(itCls->clusterSize)));
+    // LogDebug ("rpssimsetup")<<"Filling cls in _mapDetClsMapLegacy[detId,clsVect] :: detId = "<<detId;
+    // LogDebug ("rpssimsetup")<<" --> will it be accepted? clsCounter = "<<clsCounter<<" accepted?";
+    // LogDebug ("rpssimsetup")<<" New Format ::"<<((!(clsCounter%120)) && (clsCounter!=0)); // <<std::endl;
+    // LogDebug ("rpssimsetup")<<" Old Format ::"<<((!(clsCounter%100)) && (clsCounter!=0)); // <<std::endl;
+    // LogDebug ("rpssimsetup")<<std::endl;
+
+    // Old Format :: same until 100
+    if((!(clsCounter%100)) && (clsCounter!=0)){
+      detId=itCls->dpid;
+      _mapDetClsMapLegacy[detId]=clsVect;
+      clsVect.clear();
+      // LogDebug ("rpssimsetup")<<" --> Old Method ";
+      // LogDebug ("rpssimsetup")<<" --> saved in map "<<std::endl;
+      // LogDebug ("rpssimsetup")<<"Filling cls in _mapDetClsMapLegacy[detId,clsVect] :: detId = "<<detId;
+      // LogDebug ("rpssimsetup")<<" --> will it be accepted? clsCounter = "<<clsCounter<<" accepted? "<<((!(clsCounter%120)) && (clsCounter!=0))<<std::endl;
+      clsCounter=0;
+    }
+    /*else{
+      LogDebug ("rpssimsetup")<<" --> not saved in map "<<std::endl;
+      } */
+    ++clsCounter;
+  }
+  // ###########################################################################
+  LogDebug ("rpssimsetup")<<"RPCSimSetUp::setRPCSetUp :: ClusterSizeItem :: end"<<std::endl;
+
+  LogDebug ("rpssimsetup")<<"RPCSimSetUp::setRPCSetUp :: NoiseItem :: begin"<<std::endl;
+  // ### NoiseItem #############################################################
+  unsigned int count_strips = 1;
+  unsigned int count_all    = 1;
+  std::vector<float> vveff, vvnoise;
+  // vveff.clear();
+  // vvnoise.clear(); 
+
+  // DetId to start with needs to be a DetId inside the Geometry used
+  // Therefore loop on the NoiseItems and search for the first valid roll in the Geometry
+  // Assign this as the DetId to start with (so called current_roll) and quit the loop
+  bool quitLoop = false;
+  current_detId = 0; current_nStrips = 0; // current_rpcId = 0; current_roll = 0;
+  for(std::vector<RPCStripNoises::NoiseItem>::const_iterator it = vnoise.begin(); it != vnoise.end() && !quitLoop; ++it) {
+    // roll associated to the conditions of this strip (iterator)
+    current_detId = it->dpid;
+    current_rpcId = RPCDetId(current_detId);
+    // Test whether this roll (picked up from the conditions) is inside the RPC Geometry
+    const RPCRoll* roll = dynamic_cast<const RPCRoll* >(theGeometry->roll(current_rpcId));
```

Comment:
theGeometry here is RPCGeometry
in RPCGeometry
```C++
  const RPCRoll* roll(RPCDetId id) const;
```
what's the point of dynamic casts?